dds/DCPS/transport/framework/DataLinkSet.cpp

```diff
@@ -107,6 +107,19 @@ void OpenDDS::DCPS::DataLinkSet::terminate_send_if_suspended()
   }
 }
 
+OpenDDS::DCPS::SequenceNumber OpenDDS::DCPS::DataLinkSet::cur_cumulative_ack(const GUID_t& writer_id,
+                                                                             const GUID_t& reader_id) const
+{
+  SequenceNumber sn = SequenceNumber::ZERO();
+
+  GuardType guard(this->lock_);
+  for (MapType::const_iterator pos = map_.begin(), limit = map_.end(); pos != limit; ++pos) {
+    sn = std::max(sn, pos->second->cur_cumulative_ack(writer_id, reader_id));
```

Comment:
What's the meaning of the maximum cumulative ack across multiple DataLinks?  It seems like there would be only one DataLink that handles this writer/reader pair -- maybe add a comment?